### PR TITLE
feat: Localize dapp urls - MEED-2132 - Meeds-io/MIPs#48 

### PR DESCRIPTION
**Pull request by MayTekayaa**
This change will localize all dapp urls to improve referencing and to help analytics.

### CODE REVIEW

File: deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java [hunk 1]
```diff
@@ -124,17 +129,28 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
...
+      }
+      String uri = servletPath;
+      if(!uri.startsWith("/fr/") && (DAPP_PATHS_FR_UNCOM.contains(uri) || STATIC_PATHS_FR.contains(uri))) {
+        uri = "/fr" + uri;
+        response.setHeader("Location", uri);
+        response.setStatus(HttpServletResponse.SC_MOVED_PERMANENTLY);
+        return;
+      }
+      if(uri.startsWith("/fr/")) {
+        uri = uri.substring(3, uri.length());
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java]:
```suggestion
      if(uri.startsWith("/fr/")) {
        uri = uri.substring(3, uri.length());
      } else if(DAPP_PATHS_FR_UNCOM.contains(uri) || STATIC_PATHS_FR.contains(uri)) {
        uri = "/fr" + uri;
        response.setHeader("Location", uri);
        response.setStatus(HttpServletResponse.SC_MOVED_PERMANENTLY);
        return;
      }
```

File: deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java [hunk 2]
```diff
@@ -189,7 +205,7 @@ private String getPageHeaderMetadataContent(HttpServletRequest request, String s
 
   private String getLanguage(HttpServletRequest request) {
     String path = request.getServletPath();
-    return DAPP_PATHS_FR.contains(path) || STATIC_PATHS_FR.contains(path) ? "fr" : "en";
+    return path.contains("/fr") ? "fr" : "en";
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java]:
```suggestion
    return path.startsWith("/fr/") ? "fr" : "en";
```

File: deeds-dapp-webapp/src/main/webapp/WEB-INF/jsp/placeholder.jsp [hunk 3]
```diff
@@ -1,50 +1,51 @@
 <%@page import="io.meeds.deeds.web.utils.Utils"%>
 <%
   String buildnumber = Utils.getApplicationBuildNumber();
+  String basePath = request.getServletPath().startsWith("/fr/") ? "../" : "./";
```
Comment (boubaker) [inline deeds-dapp-webapp/src/main/webapp/WEB-INF/jsp/placeholder.jsp]: No need to make this change on placeholder

```suggestion
```

File: deeds-dapp-webapp/src/main/webapp/sitemap.xml [hunk 4]
```diff
@@ -4,6 +4,11 @@
     <lastmod>2023-04-17</lastmod>
     <priority>1.00</priority>
   </url>
+  <url>
+    <loc>https://www.meeds.io/home</loc>
+    <lastmod>2023-04-17</lastmod>
+    <priority>1.00</priority>
+  </url>
```
Comment (boubaker) [inline deeds-dapp-webapp/src/main/webapp/sitemap.xml]: The same page is added twice in Sitemap, I think it's not recommended, can you verify this information please.

```suggestion
  <url>
    <loc>https://www.meeds.io/home</loc>
    <lastmod>2023-04-17</lastmod>
    <priority>1.00</priority>
  </url>
```

File: deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java (hunk 1 above)
Comment (MayTekayaa) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java]: done!

File: deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java (hunk 2 above)
Comment (MayTekayaa) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java]: changed! with  path.startsWith("/fr") (without "/" at the end) , for the case of  home page in fr

File: deeds-dapp-webapp/src/main/webapp/WEB-INF/jsp/placeholder.jsp (hunk 3 above)
Comment (MayTekayaa) [reply deeds-dapp-webapp/src/main/webapp/WEB-INF/jsp/placeholder.jsp]: done!

File: deeds-dapp-webapp/src/main/webapp/sitemap.xml (hunk 4 above)
Comment (MayTekayaa) [reply deeds-dapp-webapp/src/main/webapp/sitemap.xml]: fixed!

**Review by boubaker — APPROVED** [review]

File: deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java (hunk 2 above)
Comment (boubaker) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java]: ok, then it should be like this, to be more consistent in fact

```suggestion
    return StringUtils.equals(path, "/fr") || path.startsWith("/fr/") ? "fr" : "en";
```

Comment (MayTekayaa) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/web/filter/RequestDispatcherFilter.java]: done!

**Review by boubaker — APPROVED** [review]

**Pull request merged**